examples/src/main/scala/src/main/scala/za/co/absa/enceladus/examples/CustomRuleSample4.scala

```diff
@@ -0,0 +1,186 @@
+      .save(path)
+  }
+
+
+  private def getCmdLineArguments(args: Array[String]): CmdConfigLocal = {
+    val parser = new CmdParser("spark-submit [spark options] CustomRuleSample4.jar")
+
+    val optionCmd = parser.parse(args, CmdConfigLocal())
+    if (optionCmd.isEmpty) {
+      // Wrong arguments provided, the message is already displayed
+      System.exit(1)
+    }
+    optionCmd.get
+  }
+
+
+  def main(args: Array[String]): Unit = {
+    val cmdConfigLocal: CmdConfigLocal = getCmdLineArguments(args)
+    println(CmdConfigLocal)
+
+    implicit val spark: SparkSession = SparkSession.builder
+      .master("local[*]")
+      .appName("CustomRuleSample4")
+      .config("spark.sql.codegen.wholeStage", value = false)
+      .getOrCreate()
+
+    // Do not display INFO entries for tests
+    Logger.getLogger("org").setLevel(Level.WARN)
+    Logger.getLogger("akka").setLevel(Level.WARN)
+    spark.sparkContext.setLogLevel("WARN")
+
+    import spark.implicits._
+
+    implicit val progArgs: CmdConfig = CmdConfig() // here we may need to specify some parameters (for certain rules)
+    implicit val dao: EnceladusDAO = EnceladusRestDAO // you may have to hard-code your own implementation here (if not working with menas)
+    implicit val enableCF: Boolean = false
+
+
+    var dfReader: DataFrameReader = {
+      val dfReader0 = spark.read
+      val dfReader1 = if (cmdConfigLocal.rowTag.isDefined) dfReader0.option("rowTag", cmdConfigLocal.rowTag.get) else dfReader0
+      val dfReader2 = if (cmdConfigLocal.csvDelimiter.isDefined) dfReader1.option("delimiter", cmdConfigLocal.csvDelimiter.get) else dfReader1
+      val dfReader3 = if (cmdConfigLocal.csvHeader.isDefined) dfReader2.option("header", cmdConfigLocal.csvHeader.get) else dfReader2
+      dfReader3
+    }
+
+    val inputData: DataFrame = cmdConfigLocal.inputFormat.toLowerCase match {
+      case "csv" => dfReader.csv(cmdConfigLocal.inputFile)
+      case "xml" => dfReader.format("com.databricks.spark.xml").load(cmdConfigLocal.inputFile)
+      case "parquet" => dfReader.parquet(cmdConfigLocal.inputFile)
+      case "json" => dfReader.json(cmdConfigLocal.inputFile)
+      case _ => throw new Exception("Unsupported input format")
+    }
+
+    val conformanceDef =  Dataset(
+      name = "Custom rule sample 4",
+      version = 0,
+      hdfsPath = "/a/b/c",
+      hdfsPublishPath = "/publish/a/b/c",
+
+      schemaName = "Not really used here",
+      schemaVersion = 9999,
+
+      conformance = List(
+        UppercaseCustomConformanceRule(order = 0, outputColumn = "upper", controlCheckpoint = false, inputColumn = "text_column"),
+        LPadCustomConformanceRule(order = 1, outputColumn = "final", controlCheckpoint = false, inputColumn = "upper", len = 25, pad = ".")
+      )
+    )
+
+    val outputData: DataFrame = DynamicInterpreter.interpret(conformanceDef, inputData)
+
+    outputData.show()
+    saveToCsv(outputData, cmdConfigLocal.outPath)
+
+    spark.stop()
```

Comment:
You don't need to call `spark.stop()` as it is registered by default as a shutdownhook.